drivers/ukrandom/lcpu/init.c

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/errptr.h>
+#include <uk/random/driver.h>
+
+#if CONFIG_LIBUKBOOT
+#include <uk/boot/earlytab.h>
+#endif /* CONFIG_LIBUKBOOT */
+
+struct uk_random_driver_ops *device_init(void);
+
+struct uk_random_driver driver = {
+	.name = "CPU"
+};
+
+static int uk_random_lcpu_init(struct ukplat_bootinfo __unused *bi)
+{
+	int rc;
+
+	/* initialize the device */
+	driver.ops = device_init();
+	if (unlikely(PTRISERR(driver.ops))) {
+		uk_pr_err("Could not initialize the HWRNG (%d)\n",
+			  PTR2ERR(driver.ops));
+		return PTR2ERR(driver.ops);
+	}
+
+	/* register with libukrandom */
+	rc = uk_random_init(&driver);
+	if (unlikely(rc)) {
+		uk_pr_err("Could not register with libukrandom (%d)\n", rc);
+		return rc;
+	}
+
+	return 0;
+}
+
+#if CONFIG_LIBUKBOOT
```

Comment:
The motivation was to leave `init()` available for platforms that may not wish to use `libukboot`, so that they can still initialize the library by calling `init()` directly. In that case, we can address the warning by adding `__maybe_unused`. Overall I don't have a strong opinion, as we will need to formalize the behavior of library initialization in the scope of `libukboot` rearch anyway.

Reply:
Okay, in such a case `uk_random_lcpu_init()` should not be declared static and we should provide it via `exportsyms.uk`. Secondly, can't we ignore the input parameter and just declare it with unspecified parameters?: `int uk_random_lcpu_init()`. This would look more compatible to non-`libukboot` builds. Additionally we would need to provide the function prototype with a driver-specific header, like `uk/ukrandom/lcpu/lcpu.h`. We can then decide if we make it only available for non-`libukboot` builds which I guess would make sense.

Reply:
Okay, since we don't have a final decision on the behavior of drivers when it comes to that aspect I'll follow your initial suggestion and make compilation of that file conditional to libukboot.